tests/cypress/tests-e2e/3d/layers.cy.js

```diff
@@ -184,13 +184,15 @@ describe('Test of layer handling in 3D', () => {
         cy.clickOnMenuButtonIfMobile()
         cy.get('[data-cy="3d-button"]').click()
         cy.waitUntilCesiumTilesLoaded()
-        cy.wait('@terrain-3d-layer')
         cy.readWindowValue('cesiumViewer').then((viewer) => {
-            const mainCollection = viewer.scene.primitives.get(0)
-            expect(mainCollection.length).to.eq(1)
-            const layerCollection = mainCollection.get(0)
+            // main collection
+            cy.wrap(viewer.scene.primitives.get(0)).should('have.length', 1)
+            // layer collection
             // should be 3 (line, icon, text) but ol-cesium creates additional empty collection
-            expect(layerCollection.length).to.eq(4)
+            cy.wrap(viewer.scene.primitives.get(0).get(0), { timeout: 10000 }).should(
```

Comment:
so, it's kind of waiting until the timeout or the length is 4, right?

Reply:
@ismailsunni yes its correct using the `cy.wrap()` make use of cypress retriability and it a best practice in comparison to the `expect` depending on the use case. However it turned out that here that was not the issue. Yesterday I did a deeper debugging session with Pascal and it seems that we have a race condition in our code and that effectively sometimes on the cypress test the primitives of the KML are missing and the KML is not displayed. We could not find out the real reason and we skip that test for the moment. This not so critical as it works and for the go live this feature (display of KML in 3d is not a stopper as it is a new feature not available on the old viewer.
See #643